packages/contracts/test/token/erc20/governance-wrapped-erc20.ts

```diff
@@ -0,0 +1,335 @@
+    );
+    await Promise.all(promises);
+
+    defaultGovernanceWrappedERC20InitData = [
+      erc20.address,
+      governanceWrappedERC20Name,
+      governanceWrappedERC20Symbol,
+    ];
+  });
+
+  describe('initialize:', async () => {
+    it('reverts if trying to re-initialize', async () => {
+      erc20 = await TestERC20.deploy(...defaultExistingERC20InitData);
+
+      governanceToken = await GovernanceWrappedERC20.deploy(
+        ...defaultGovernanceWrappedERC20InitData
+      );
+
+      await expect(
+        governanceToken.initialize(...defaultGovernanceWrappedERC20InitData)
+      ).to.be.revertedWith(OZ_ERRORS.ALREADY_INITIALIZED);
+    });
+
+    it('sets the wrapped token name and symbol', async () => {
+      governanceToken = await GovernanceWrappedERC20.deploy(
+        ...defaultGovernanceWrappedERC20InitData
+      );
+
+      expect(await governanceToken.name()).to.eq(governanceWrappedERC20Name);
+      expect(await governanceToken.symbol()).to.eq(
+        governanceWrappedERC20Symbol
+      );
+    });
+  });
+
+  describe('supportsInterface:', async () => {
+    it('it supports all inherited interfaces', async () => {
+      governanceToken = await GovernanceWrappedERC20.deploy(
+        ...defaultGovernanceWrappedERC20InitData
+      );
+
+      await Promise.all(
+        [
+          'IGovernanceWrappedERC20',
+          'IERC165Upgradeable',
+          'IERC20Upgradeable',
+          'IERC20PermitUpgradeable',
+          'IVotesUpgradeable',
+        ].map(async interfaceName => {
+          const iface = new ethers.utils.Interface(
+            // @ts-ignore
+            (await hre.artifacts.readArtifact(interfaceName)).abi
+          );
+          expect(await governanceToken.supportsInterface(getInterfaceID(iface)))
+            .to.be.true;
+        })
+      );
+      // We must check `IERC20MetadataUpgradeable` separately as it inherits from `IERC20Upgradeable` and we cannot get the isolated ABI.
+      const ierc20MetadataInterface = new ethers.utils.Interface([
+        'function name()',
+        'function symbol()',
+        'function decimals()',
+      ]);
+      expect(
+        await governanceToken.supportsInterface(
+          getInterfaceID(ierc20MetadataInterface)
+        )
+      ).to.be.true;
+    });
+  });
+
+  describe('depositFor', async () => {
+    beforeEach(async function () {
+      governanceToken = await GovernanceWrappedERC20.deploy(
+        ...defaultGovernanceWrappedERC20InitData
+      );
+    });
+
+    it('reverts if the amount is not approved', async () => {
+      const erc20Balance = await erc20.balanceOf(signers[0].address);
+      await expect(
+        governanceToken.depositFor(signers[0].address, erc20Balance)
+      ).to.be.revertedWith(OZ_ERRORS.ERC20_INSUFFICIENT_ALLOWANCE);
+    });
+
+    it('deposits an amount of tokens', async () => {
+      // check old balances
+      const erc20Balance = await erc20.balanceOf(signers[0].address);
```

Comment:
I'd do `ownerAddress` here. It becomes handy later once you quickly need it, instead of writing `signers[0]`